core/src/main/scala/org/bykn/bosatsu/rankn/Infer.scala

```diff
@@ -1231,10 +1381,65 @@ object Infer {
             case notMatch => notMatch
           }
 
-          (checkSigma(inner, tpe), instSigma(tpe, expect, region(tag)))
-            .parFlatMapN { (typedTerm, coerce) =>
-              zonkTypedExpr(typedTerm).map(coerce(_))
-            }
+          @inline def default = checkAnnotated(inner, tpe, region(tag), expect)
+
+          tpe match {
+            case rho: Type.Rho =>
```

Comment:
seems like this should only check `Type.Rho` when expect is Infer. Otherwise we can use `subsCheck`.

Secondly, it seems like this whole algorithm should actually be inside `checkSigma` which is logically what it is doing, but checkSigma is called in some other places.